File: 3914-check-if-any-element-has-prime-frequency/3914-check-if-any-element-has-prime-frequency.java
class Solution {
    boolean pr[]=new boolean[101];
    private void primenumber(){
        for(int i=2;i<101;i++){
            pr[i]=true;
        }
        for(int p=2;p*p<101;p++){
            if(pr[p]==true){
                for(int i=p*p;i<101;i+=p){
                    pr[i]=false;
                }
            }
        }
    }
    public boolean checkPrimeFrequency(int[] nums) {
        primenumber();
        int i,j,n=nums.length;
        int[] freq=new int[101];
        for(i=0;i<n;i++){
            freq[nums[i]]++;
        }
        for(i=0;i<101;i++){
            if(freq[i]!=0){
                if(pr[freq[i]])
                    return true;
            }
        }
        return false;
    }
}

File: 3914-check-if-any-element-has-prime-frequency/3914-check-if-any-element-has-prime-frequency.cpp
class Solution {
public:
    bool isPrime(int n) {
        if (n <= 1)
            return false;
        if (n <= 3)
            return true;
        if (n % 2 == 0 || n % 3 == 0)
            return false;
        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0) {
                return false;
            }
        }
        return true;
    }

    bool checkPrimeFrequency(vector<int>& nums) {
        unordered_map<int, int> hash;
        for (auto it : nums) {
            hash[it]++;
        }
        for (auto it : hash) {
            if (isPrime(it.second)) {
                return true;
            }
        }
        return false;
    }
};